online-inference/hf-llm/service/service.py

```diff
@@ -135,8 +144,10 @@ def load_model(path_uri: str) -> TextGenerationPipeline:
     logger.info(f"Loading {args.precision} model from: {path_uri}")
     if path_uri.startswith("s3://"):
         return load_model_s3(path_uri)
-    else:
+    elif os.path.exists(f"{path_uri}/model.tensors"):
         return load_model_local(path_uri)
+    else:
+        return load_model_hub(path_uri)
```

Comment:
The `s3://` case accounts for providing a fully specified path such as `s3://tensorized/EleutherAI/pythia-70m/fp16/model.tensors`; this case should do something similar and check if the referred-to path is a file or a directory, e.g.

```suggestion
    if os.path.exists(path_uri):
        if os.path.isfile(path_uri):
            return load_pipeline(path_uri)
        full_path: str = os.path.join(path_uri, "model.tensors")
        if os.path.isfile(full_path):
            return load_pipeline(full_path)
    return load_model_hub(path_uri)
```

This leaves `load_model_hub` as the loading path for `path_uri`s that either:
- Do not exist on the filesystem, or
- Are directories that don't contain a `model.tensors` file

(This series of conditionals could also be written a bit more cleanly with `pathlib`)